### PR TITLE
Regenerated yarn.lock file and upgrade VSCE to 3.5.0

### LINKED ISSUE #1285: Update Mocha to Adress Vulnerabilities
Uses of Mocha:

Image

It appears to be used for our test cases. 

### Validation
Criteria for validating if upgrade is successful if running `yarn test` results in similar previous behavior.

### PR DESCRIPTION
**Pull request by JeremyYao**
image

Closes #1285, #1286, and #1287

### How to Test

Follow intensive manual testing checklist: https://github.com/apache/daffodil-vscode/blob/main/src/tests/README.md

### CODE REVIEW

**Comment by JeremyYao** [comment]
Seems to break copy TDML functionaltiy and append 
image

**Comment by JeremyYao** [comment]
Added testing instructions to this ticket. Please follow intensive testing checklist https://github.com/apache/daffodil-vscode/blob/main/src/tests/README.md as well as performing testing that is representative of your usage of the extension. Goal of this is to catch if dependencies has caused capabilities to break and edge cases.

**Review by rthomas320 — APPROVED** [review]
+1

File: package.json [hunk 1]
```diff
@@ -84,7 +84,7 @@
     "fast-xml-parser": "^4.5.2",
     "glob": "8.1.0",
     "mini-css-extract-plugin": "^2.7.6",
-    "mocha": "10.2.0",
+    "mocha": "10.8.2",
```
Comment (shanedell) [inline package.json:87]:
```suggestion
    "mocha": "11.5.0",
```

This [dependabot PR](https://github.com/apache/daffodil-vscode/pull/1278) updates the version to 11.5.0, so is this PR necessary or should we use the dependabot one?

**Review by shanedell — CHANGES_REQUESTED** [review]

File: yarn.lock [hunk 2]
```diff
@@ -800,65 +796,65 @@
...
+    "@vscode/vsce-sign-alpine-x64" "2.0.5"
+    "@vscode/vsce-sign-darwin-arm64" "2.0.5"
+    "@vscode/vsce-sign-darwin-x64" "2.0.5"
+    "@vscode/vsce-sign-linux-arm" "2.0.5"
+    "@vscode/vsce-sign-linux-arm64" "2.0.5"
+    "@vscode/vsce-sign-linux-x64" "2.0.5"
+    "@vscode/vsce-sign-win32-arm64" "2.0.5"
+    "@vscode/vsce-sign-win32-x64" "2.0.5"
 
 "@vscode/vsce@3.2.2":
```
[the remark this answers is not in the document]
Comment (JeremyYao) [reply yarn.lock]: Thanks for the heads up! I went ahead and upgraded the vsce package to 3.5.0 and regenerated the yarn.lock file. 

Let's go ahead and try the newer/simpler workflow after this gets merged in?

[the remark this answers is not in the document]
Comment (stevedlawrence) [reply yarn.lock]: @shanedell, do you want to use the new release candidate process for rc2?

[the remark this answers is not in the document]
Comment (shanedell) [reply yarn.lock]: @stevedlawrence yes I would like to do that rc2

[the remark this answers is not in the document]
Comment (stevedlawrence) [reply yarn.lock]: :+1: I've enabled the "[Release Candidate](https://github.com/apache/daffodil-vscode/actions/workflows/release-candidate.yml)" github action used by the new workflow. If you change your mind and decide to follow the old steps make sure to first disable that workflow or let me know and I can disable it.
